zerver/views/user_groups.py

```diff
@@ -107,6 +107,48 @@ def get_user_group(request: HttpRequest, user_profile: UserProfile) -> HttpRespo
     return json_success(request, data={"user_groups": user_groups})
 
 
+def get_parsed_group_setting_value(user_group: UserGroup) -> Union[int, AnonymousSettingGroupDict]:
+    if hasattr(user_group, "named_user_group"):
+        return user_group.id
+
+    return {
+        "direct_members": list(user_group.direct_members.all().values_list("id", flat=True)),
+        "direct_subgroups": list(user_group.direct_subgroups.all().values_list("id", flat=True)),
+    }
+
+
+def is_api_value_same_as_current_setting_value(
+    current_value: Union[int, AnonymousSettingGroupDict],
+    api_value: Union[int, AnonymousSettingGroupDict],
+) -> bool:
+    if isinstance(current_value, int):
+        if isinstance(api_value, int):
+            return api_value == current_value
+
+        if len(api_value["direct_members"]) == 0 and len(api_value["direct_subgroups"]) == 1:
+            return api_value["direct_subgroups"][0] == current_value
+
+        return False
+
+    if isinstance(api_value, int):
+        return False
+
+    return set(api_value["direct_members"]) == set(current_value["direct_members"]) and set(
+        api_value["direct_subgroups"]
+    ) == set(current_value["direct_subgroups"])
+
+
+def check_setting_value_changed(
+    current_value: UserGroup,
+    new_setting_value: Union[int, AnonymousSettingGroupDict],
+) -> bool:
+    parsed_current_setting_value = get_parsed_group_setting_value(current_value)
+
+    return not is_api_value_same_as_current_setting_value(
+        parsed_current_setting_value, new_setting_value
```

Comment:
I am not sure of the function and variable name here, just added them in a way so that we can avoid duplicate logic and also need not have queries like `user_group.direct_members.all()` twice. Any suggestions would be helpful here.